pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -185,8 +190,10 @@ private String getMultiStageQueryResponse(String query, String queryOptions, Htt
       return QueryException.ACCESS_DENIED_ERROR.toString();
     }
 
-    SqlNodeAndOptions sqlNodeAndOptions = RequestUtils.parseQuery(query);
-    List<String> tableNames = CalciteSqlParser.extractTableNamesFromNode(sqlNodeAndOptions.getSqlNode());
```

Comment:
this can be removed from `CalciteSqlParser.extractTableNamesFromNode`

Reply:
Hey @walterddr i think we are still using this method in pinot-java-client which got introduced in #10692.

Reply:
hmm. ok. that is harder to incorporate b/c client doesnt really have full access to table names. 
CC @abhioncbr

Reply:
LGTM.

Reply:
what's your thought on this @abhioncbr ? do you think we have a better way to handle table look up? ideally speaking shouldn't keep this hacky SqlNode parser level table finder b/c there's corner cases we can't really do on the syntactic parsing level.
What i can think about is 
- only allow simple broker selector if dealing with V2 engine queries for now. (so there's no need to extract table name, we blindly select a broker)
- we should support controller based broker selector to ask controller to return the broker --> this way we avoid the table metadata problem b/c controller have access to all table metadata.

Thoughts?

Reply:
i dont know if there's a SqlParser level util that can do that in calcite. but yes i think for now we can do the single broker selector.

Reply:
Hey @walterddr raised this #10955 to make the change in Pinot client to keep it independent of this change. Will remove the Calcite-util in this PR or the other one which one will merge later. Hope so it's okay! 

Looking into the integration test failure of this one meanwhile.

Reply:
sounds good thank you will take another look at #10955 as well.